`Creación de Variables`

##### Importar paquetes

In [106]:
#importamos libreria
import pandas as pd
import numpy as np

##### Cargar datos

In [107]:
# Ruta al archivo CSV en la carpeta data
file_path = '../data/online_course_engagement_data.csv'

# Leer el archivo CSV
df = pd.read_csv(file_path)

# Borramos datos duplicados y creamos un nuevo df1
df1 = df.drop_duplicates(keep='first')
df1 = df1.copy()

1. Estadísticas de finalización de cursos

In [108]:
# Las estadísticas descriptivas para la columna CourseCompletion son las siguientes:
df1['CourseCompletion'].describe()


count    8123.000000
mean        0.439247
std         0.496326
min         0.000000
25%         0.000000
50%         0.000000
75%         1.000000
max         1.000000
Name: CourseCompletion, dtype: float64

*Conclusion: Estos resultados indican que aproximadamente el 43.9% de los cursos fueron completados, con una alta variabilidad (casi 50%).*

*Luego Creamos nuevo dataframe (df2) que contiene aquellos usuarios que completarón el curso, de acuerdo a los registros.*

In [109]:
df2 = df1[df1['CourseCompletion'] == 1]
print(f'El total de registros de este nuevo df2 es:', len(df2))

El total de registros de este nuevo df2 es: 3568


2. Calculo de las Puntuaciones de los cuestionarios, primero para el conjunto de datos df1, luego df2

In [110]:
#df1
#Ayuda a entender el rendimiento general de los usuarios en las evaluaciones del curso.
#De las pruebas realizadas por los usuarios del curso evaluamos el valor promedio, maximo y 
#minimo de las puntuaciones. 

AverageQuizScore=df1['QuizScores'].mean()

print(f'El Promedio de Puntuación de los Cuestionarios para df1 es:', AverageQuizScore.round(2))

El Promedio de Puntuación de los Cuestionarios para df1 es: 75.09


In [111]:
#df2
# Ayuda a entender el rendimiento general de los usuarios en las evaluaciones del curso.
#De las pruebas realizadas por los usuarios del curso evaluamos el valor promedio, maximo y 
#minimo de las puntuaciones. 

AverageQuizScore2=df2['QuizScores'].mean()

print(f'El Promedio de Puntuaciones de los Cuestionarios para df2 es:', AverageQuizScore2.round(2))

El Promedio de Puntuaciones de los Cuestionarios para df2 es: 80.03


2. Tiempo Promedio por Video para cada conjunto de datos (df1 y df2)

In [112]:
#df1
#eliminamos resultados infinitos, donde el 'NumberOfVideosWatched' es 0
df1_video = df1[df1['NumberOfVideosWatched'] != 0]
df1_video = df1_video.copy()
df1_video.loc[:, 'AverageTimePerVideo'] = df1_video['TimeSpentOnCourse'] / df1_video['NumberOfVideosWatched']

#tiempo promedio en horas que un usuario pasa viendo cada video.
df1_video.loc[:, 'AverageTimePerVideo'] = df1_video['TimeSpentOnCourse'] / df1_video['NumberOfVideosWatched']
print((df1_video['AverageTimePerVideo'].describe()).round(2))

count    7776.00
mean        8.73
std        12.89
min         0.05
25%         2.49
50%         4.83
75%         8.84
max        99.62
Name: AverageTimePerVideo, dtype: float64


In [113]:
#df2
#eliminamos resultados infinitos, donde el 'NumberOfVideosWatched' es 0
df2_video = df2[df2['NumberOfVideosWatched'] != 0]
df2_video = df2_video.copy()
df2_video.loc[:, 'AverageTimePerVideo'] = df2_video['TimeSpentOnCourse'] / df2_video['NumberOfVideosWatched']

#tiempo promedio en horas que un usuario pasa viendo cada video.
df2_video.loc[:, 'AverageTimePerVideo'] = df2_video['TimeSpentOnCourse'] / df2_video['NumberOfVideosWatched']
print((df2_video['AverageTimePerVideo'].describe()).round(2))

count    3507.00
mean        7.02
std         9.70
min         0.05
25%         2.93
50%         4.74
75%         7.23
max        97.99
Name: AverageTimePerVideo, dtype: float64


3. Tiempo Promedio por Cuestionario para cada conjunto de datos (df1 y df2)

In [114]:
#df1
#eliminamos resultados infinitos, donde el 'NumberOfQuizzesTaken' es 0
df1_cuestionario = df1[df1['NumberOfQuizzesTaken'] != 0]
df1_cuestionario = df1_cuestionario.copy()

#Calcula el tiempo promedio que un usuario pasa por cuestionario.
df1_cuestionario.loc[:, 'AverageTimePerQuiz'] = (df1_cuestionario['TimeSpentOnCourse'] / df1_cuestionario['NumberOfQuizzesTaken']).round(2)
print((df1_cuestionario['AverageTimePerQuiz'].describe()).round(2))

count    7475.00
mean       14.30
std        16.71
min         0.10
25%         4.66
50%         8.99
75%        16.34
max        99.99
Name: AverageTimePerQuiz, dtype: float64


In [115]:
#df2
#eliminamos resultados infinitos, donde el 'NumberOfQuizzesTaken' es 0
df2_cuestionario = df2[df2['NumberOfQuizzesTaken'] != 0]
df2_cuestionario = df2_cuestionario.copy()

#Calcula el tiempo promedio que un usuario pasa por cuestionario.
df2_cuestionario.loc[:, 'AverageTimePerQuiz'] = (df2_cuestionario['TimeSpentOnCourse'] / df2_cuestionario['NumberOfQuizzesTaken']).round(2)
print((df2_cuestionario['AverageTimePerQuiz'].describe()).round(2))

count    3448.00
mean       11.92
std        12.65
min         0.11
25%         5.42
50%         8.86
75%        13.66
max        99.99
Name: AverageTimePerQuiz, dtype: float64


*existe una diferencia de 120 usuarios que no tomaron quiz pero si completarón el curso, lo siguiente expone el autor de los datos: Algunos cursos no tienen cuestionarios, por lo que la puntuación del cuestionario generalmente se basa en otras métricas, como la cantidad de videos que se vieron.*

4. Nivel de Actividad del Usuario

In [123]:
df1["TimeSpentOnCourse"].value_counts

<bound method IndexOpsMixin.value_counts of 0       29.979719
1       27.802640
2       86.820485
3       35.038427
4       92.490647
          ...    
8995    37.445225
8996    48.631443
8997    38.212512
8998    70.048665
8999    93.589781
Name: TimeSpentOnCourse, Length: 8123, dtype: float64>

In [118]:
# Definir la función para clasificar el nivel de actividad
def ActivityLevel(row):
    if row['TimeSpentOnCourse'] < 26:
        return "Bajo"
    elif row['TimeSpentOnCourse'] < 50:
        return "Medio"
    elif row['TimeSpentOnCourse'] >= 50:
        return "Alto"

# Aplicar la función al DataFrame
df1["ActivityLevel"] = df1.apply(ActivityLevel, axis=1)

# Imprimir los totales por nivel de actividad
print(f'El total con nivel de actividad Bajo es:', len(df1.loc[df1['ActivityLevel'] == 'Bajo']))
print(f'El total con nivel de actividad Medio es:', len(df1.loc[df1['ActivityLevel'] == 'Medio']))
print(f'El total con nivel de actividad Alto es:', len(df1.loc[df1['ActivityLevel'] == 'Alto']))


El total con nivel de actividad Bajo es: 2016
El total con nivel de actividad Medio es: 2016
El total con nivel de actividad Alto es: 4091


In [124]:
# Definir los límites para los grupos
bins = [0, 26, 50, df1['TimeSpentOnCourse'].max()]
labels = ['Bajo', 'Medio', 'Alto']

# Crear una nueva columna 'ActivityLevel' basada en los intervalos
df1['ActivityLevel'] = pd.cut(df1['TimeSpentOnCourse'], bins=bins, labels=labels, include_lowest=True)

# Generar un cuadro que muestre el conteo de cada categoría
activity_level_counts = df1['ActivityLevel'].value_counts().sort_index()

# Imprimir el cuadro de conteo de cada categoría
print(activity_level_counts)


ActivityLevel
Bajo     2016
Medio    2016
Alto     4091
Name: count, dtype: int64
